PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/org/jetbrains/plugins/azure/identity/ad/ui/RegisterApplicationInAzureAdDialog.kt

```diff
@@ -0,0 +1,176 @@
+ * to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+ * <p/>
+ * The above copyright notice and this permission notice shall be included in all copies or substantial portions of
+ * the Software.
+ * <p/>
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO
+ * THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package org.jetbrains.plugins.azure.identity.ad.ui
+
+import com.intellij.openapi.Disposable
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.ui.DialogWrapper
+import com.intellij.openapi.ui.ValidationInfo
+import com.intellij.openapi.util.Disposer
+import com.intellij.ui.components.JBTextField
+import com.intellij.ui.layout.panel
+import org.jetbrains.plugins.azure.RiderAzureBundle
+import org.jetbrains.plugins.azure.isValidUrl
+import javax.swing.JLabel
+
+class RegisterApplicationInAzureAdDialog(project: Project, val model: RegistrationModel)
+    : DialogWrapper(project), Disposable {
+
+    class RegistrationModel(
+            val originalDisplayName: String,
+            val originalClientId : String,
+            val originalDomain : String,
+            val originalCallbackUrl : String,
+            val originalIsMultiTenant : Boolean,
+            var allowOverwrite : Boolean
+    ) {
+        var updatedDisplayName: String = originalDisplayName
+        var updatedDomain : String = originalDomain
+        var updatedClientId : String = originalClientId
+        var updatedCallbackUrl : String = originalCallbackUrl
+        var updatedIsMultiTenant : Boolean = originalIsMultiTenant
+        var hasChanges : Boolean = false
+    }
+
+    private val panel = panel {
+
+        noteRow(RiderAzureBundle.message("dialog.identity.ad.register_app.description"))
+
+        row {
+            val label = JLabel(RiderAzureBundle.message("dialog.identity.ad.register_app.display_name"))
+            label()
+
+            cell {
+                textField({ model.originalDisplayName },
+                        {
+                            model.updatedDisplayName = it
+                            model.hasChanges = true
+                        })
+                        .withValidationOnInput { validateNotEmpty(it) }
+                        .component.apply { label.labelFor = this }
+            }
+        }
+
+        row {
+            val label = JLabel(RiderAzureBundle.message("dialog.identity.ad.register_app.callback_url"))
+            label()
+
+            cell {
+                textField({ model.originalCallbackUrl },
```

Comment:
The default Callback URL is loaded when I change `appsettings.json` and reopen the dialog to re-register an app.

Reply:
Fixed.